stack/stack.go

```diff
@@ -44,25 +44,25 @@ func (s S) String() string {
 	return s.Name()
 }
 
-func IsLeaf(dir string) (bool, error) {
-	l := NewLoader()
+func IsLeaf(root, dir string) (bool, error) {
+	l := NewLoader(root)
 	return l.IsLeafStack(dir)
 }
 
-func LookupParent(dir string) (S, bool, error) {
-	l := NewLoader()
+func LookupParent(root, dir string) (S, bool, error) {
+	l := NewLoader(root)
 	return l.lookupParentStack(dir)
 }
 
 // Load a single stack from dir.
-func Load(dir string) (S, error) {
-	l := NewLoader()
+func Load(root, dir string) (S, error) {
```

Comment:
Im thinking that for path manipulation stuff maybe the stack.S could keep the root inside and be able to provide all sorts of different stack paths required (I think this is spread in multiple places).

Reply:
Like maybe the showdir thing, if it is more to showing stack paths in a nice/cute way, could be a method on the stack itself, like a nice string representation of the type, etc (just wondering/brainstorming)

Reply:
I liked this idea, we can improve in other PR.